docs/SETTINGS.md

```diff
@@ -0,0 +1,44 @@
+# Settings
+
+## Supported Options
+
+The language server supports the following configuration options:
+
+## `rootModulePaths` (`[]string`)
+
+This allows overriding automatic root module discovery by passing a static list
+of absolute paths to root modules (i.e. folders with `*.tf` files
+which have been `terraform init`-ed).
+
+## How to pass settings
+
+The server expects static settings to be passed as part of LSP `initialize` call,
+but how settings are requested from on the UI side depends on the client.
+
+### Sublime Text
+
+Use `initializationOptions` key under the `clients.terraform` section, e.g.
+
+```json
+{
+	"clients": {
+		"terraform": {
+			"initializationOptions": {
+				"rootModulePaths": ["/any/path"]
+			},
+		}
+	}
+}
+```
+
+### VS Code
+
+Use `terraform-ls.options`, e.g.
```

Comment:
This may need to be tweaked depending on the name we actually decide upon and ship, but if I had to choose between `configuration`, `options` or `settings`, I would choose `options`, because that reflects the naming convention in the LSP and also leaves some space in case we decide to add some more settings under `terraform-ls` which would _not_ be passed down directly to the server, but affect how it's launched or installed (e.g. version pinning comes to mind, opting into prereleases etc.).

Reply:
I was thinking it might be easier for the user to manage if the setting isn't nested, i.e. `terraform-ls.rootModules` -- but I don't know how many options we would anticipate adding later.

Reply:
> but I don't know how many options we would anticipate adding later.

I was thinking that anything version/installation/update related could go under this `terraform-ls` namespace? That way it's also unlikely to conflict with any terraform-lsp related settings, if we do decide to add any.

Also knowing how difficult/awkward it is at the moment to change/deprecate supported settings in VSCode and remembering the confusion from v1/v2 settings changes, I was thinking of applying extra caution here.

Reply:
Changed this to just `terraform-ls` as per our discussion earlier this week about how this makes it easier to expose the configuration in the UI.